map.go

```diff
@@ -1486,10 +1490,34 @@ func marshalMap(m *Map, length int) ([]byte, error) {
 	return buf, nil
 }
 
-// MapIterator iterates a Map.
+// MapIterator is the interface defining methods to iterate
+// through the content of the target map.
+//
+// See Map.Iterate.
+type MapIterator interface {
+	// Err returns any encountered error.
+	//
+	// The method must be called after Next returns nil.
+	//
+	// For key-value maps, returns ErrIterationAborted if
```

Comment:
I'd remove the 'for key-value maps' part. The interface shouldn't document the implementation. Generally, `ErrIterationAborted` means a full iteration wasn't possible.

Reply:
I agree, probably too many comments aren't needed in this case.